WooCommerce/Classes/POS/Presentation/ItemListView.swift

```diff
@@ -5,30 +5,44 @@ struct ItemListView: View {
     @ScaledMetric private var scale: CGFloat = 1.0
     @ObservedObject var viewModel: ItemListViewModel
     @Environment(\.floatingControlAreaSize) var floatingControlAreaSize: CGSize
+    @State private var showModal: Bool = false
 
     init(viewModel: ItemListViewModel) {
         self.viewModel = viewModel
     }
 
     var body: some View {
-        VStack {
-            headerView()
-            switch viewModel.state {
-            case .empty(let emptyModel):
-                emptyView(emptyModel)
-            case .loading:
-                loadingView
-            case .loaded(let items):
-                listView(items)
-            case .error(let errorModel):
-                errorView(errorModel)
+        ZStack {
+            VStack {
+                headerView()
+                switch viewModel.state {
+                case .empty(let emptyModel):
+                    emptyView(emptyModel)
+                case .loading:
+                    loadingView
+                case .loaded(let items):
+                    listView(items)
+                case .error(let errorModel):
+                    errorView(errorModel)
+                }
+            }
+            .refreshable {
+                await viewModel.reload()
+            }
+            .padding(.horizontal, Constants.itemListPadding)
+            .background(Color.posBackgroundGreyi3)
+
+            if showModal {
+                SimpleProductsModalView(isPresented: $showModal)
+                    .background(Color.clear)
+                    .transition(.move(edge: .bottom))
+                    .onAppear {
+                        withAnimation(.easeInOut) {
+                            showModal = true
+                        }
+                    }
```

Comment:
I suppose we need to show this one and other modals above the main ZStack in the POS, because now it is showing on top of products and messes up the layout. We could use one of our view models to set which modal should be shown on top of entire POS UI

![Simulator Screenshot - iPad Pro (11-inch) (4th generation) 17 2 - 2024-07-25 at 10 22 15](https://github.com/user-attachments/assets/6dff3c04-cf08-4dab-aff0-45c52f6ef6f9)

<img width="608" alt="Screenshot 2024-07-25 at 10 22 12" src="https://github.com/user-attachments/assets/718df306-0659-474b-885b-8ed18689c936">

Reply:
This has been addressed.

Reply:
> We could use one of our view models to set which modal should be shown on top of entire POS UI

Good idea. Implemented in commit 1f4084b3e9a1ea31096b73ec121555ca54e2013c.

Reply:
I needed to back this out and re-merge trunk. An earlier merge of trunk led to a tangle that I thought might need generics to sort out.

Suggest we can take up centralizing modal view presentation in a later PR.

Reply:
>We could use one of our view models to set which modal should be shown on top of entire POS UI

I added this back in commit a5df2a7be8514ddbc2d37972a7712304d2321f1d